src/Validator.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+namespace Lcobucci\JWT;
+
+use Generator;
+use Lcobucci\JWT\Claim\Validatable;
+
+/**
+ * Class that validates token with defined validationData
+ *
+ * @author Danny Dörfel <danny.dorfel@gmail.com>
+ */
+class Validator
+{
+    protected $errors = [];
+
+    /**
+     * @param Token $token
+     * @param ValidationData $data
+     * @return bool
+     */
+    public function validate(Token $token, ValidationData $data): bool
+    {
+        $this->errors = [];
+
+        foreach ($this->getValidatableClaims($token) as $claim) {
+
+            if ($data->has($claim->getName()) && !$claim->validate($data)) {
+                $this->errors[$claim->getName()] = false;
+            }
+        }
+
+        return count($this->errors) === 0;
```

Comment:
You are right and I know very well it's possible to negate the array, but I would like to advocate being explicit. Using ! on a variable would normally imply a boolean, but in PHP you can use it on a lot of different situations. For me, that is more or less like goto: just because we have goto in php doesn't mean we have to use it. Considering, using ```return empty()``` would be the best choice since we know it's an array.